File: models/creator-card.js
const { ModelSchema, SchemaTypes, DatabaseModel } = require('@app-core/mongoose');

const modelName = 'creator_cards';

/**
 * Creator Card schema. `_id` is a ULID (serialized as `id` in responses).
 * Soft-delete is handled by the paranoid option: the framework adds a `deleted`
 * field (default 0 → serialized to null), excludes deleted docs from findOne,
 * and frees the unique `slug` on delete. `created`/`updated` are set by the
 * repository create(). `links`/`service_rates` are stored as-is (Mixed) — their
 * shape is enforced on input by the VSL validator in the service layer.
 *
 * @typedef {Object} CreatorCardModel
 * @property {String} _id
 * @property {String} title
 * @property {String} description
 * @property {String} slug
 * @property {String} creator_reference
 * @property {Object[]} links
 * @property {Object} service_rates
 * @property {String} status
 * @property {String} access_type
 * @property {String} access_code
 * @property {Number} created
 * @property {Number} updated
 * @property {Number} deleted
 */

const schemaConfig = {
  _id: { type: SchemaTypes.ULID, required: true },
  title: { type: SchemaTypes.String, required: true },
  description: { type: SchemaTypes.String },
  slug: { type: SchemaTypes.String, required: true, unique: true },
  creator_reference: { type: SchemaTypes.String, required: true, index: true },
  links: { type: SchemaTypes.Mixed, required: true },
  service_rates: { type: SchemaTypes.Mixed },
  status: { type: SchemaTypes.String, required: true, index: true },
  access_type: { type: SchemaTypes.String, required: true, default: 'public' },
  access_code: { type: SchemaTypes.String },
  created: { type: SchemaTypes.Number, required: true },
  updated: { type: SchemaTypes.Number, required: true },
};

const modelSchema = new ModelSchema(schemaConfig, { collection: modelName });

/** @type {CreatorCardModel} */
module.exports = DatabaseModel.model(modelName, modelSchema, { paranoid: true });

File: specs/creator-cards/data/creator-card.go
import ../commons.go

// Creator Card — database schema (model definition, not service-input validation).
// Mongo `_id` is a ULID and is serialized as `id` in all API responses (never `_id`).
// Input validation + business rules live in the services (see ADR 0003).

CreatorCard {
  _id string<isUnique|indexed> // ULID; exposed as `id` in responses
  title string // 3–100 chars
  description string // up to 500 chars
  slug string<isUnique|indexed> // 5–50 chars, charset [A-Za-z0-9_-]; unique
  creator_reference string<indexed> // exactly 20 chars; owner secret used for delete

  // External links shown on the card
  links[] {
    title string // 1–100 chars
    url string // up to 200 chars; must start with http:// or https://
  }

  // Optional pricing block; when present, currency is set and rates is non-empty
  service_rates? {
    currency string // enum: NGN | USD | GBP | GHS
    rates[] {
      name string // 3–100 chars
      description? string // up to 250 chars
      amount number // positive integer, minor units (e.g. kobo/cents)
    }
  }

  status string<indexed> // enum: draft | published
  access_type string // enum: public | private (defaults to public)
  access_code? string // exactly 6 alphanumeric chars; required iff private, absent iff public; never returned

  ...common // created, updated, deleted?
}
